### PR TITLE
Cancel receiver pod start on invalid OIDC config only if authentication.oidc is enabled

### PR DESCRIPTION
**Pull request by creydr**
Currently we cancel the receivers pod start when the OIDC configuration could not get loaded properly (most likely the case on an invalid configuration). This happens also, when OIDC is not enabled.
This PR addresses it and only cancels the receiver pods start on an invalid config if OIDC is enabled. Otherwise it logs a warning, as this could lead to issues later, when the OIDC flag gets enabled.

### CODE REVIEW

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/oidc/TokenVerifierImpl.java
Comment (Cali0707) [inline data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/oidc/TokenVerifierImpl.java:52]: Would using the `FileWatcher` to rebuild the config solve the problem of needing to restart the pod?

Comment (creydr) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/oidc/TokenVerifierImpl.java:52]: It could. Anyhow changing something fundamental like the OIDC config will probably require some other restarts as well.

**Comment by creydr** [comment]
@Cali0707 @pierDipi could you PTAL when you find some time?
This should help us to unblock https://github.com/knative-extensions/eventing-istio/pull/262 too

**Review by Cali0707 — APPROVED** [review]
/lgtm

@creydr would it be possible to add a unit test to the `FeaturesConfig`, as a follow up (I don't think this needs to block this PR)

**Pull request merged**
